### ISSUE TITLE
Small Update to Parallel Evaluation in Jax

### ISSUE THREAD

**Issue by pharringtonp19** [google/jax#7691]
The notebook on [Parallel Evauation in Jax](https://jax.readthedocs.io/en/latest/jax-101/06-parallelism.html) states "that unlike vmap, only the leading axis (0) is supported by pmap at the time of writing this guide."

However, it seems that `pmap` now supports more than just the leading axis.

```
import jax.tools.colab_tpu
jax.tools.colab_tpu.setup_tpu()
import jax 
jax.devices()
import numpy as np 
import jax.numpy as jnp 

x = np.arange(5)
w = np.array([2., 3., 4.])

def convolve(x, w):
  output = []
  for i in range(1, len(x)-1):
    output.append(jnp.dot(x[i-1:i+2], w))
  return jnp.array(output)

n_devices = jax.local_device_count()
xs = np.arange(5 *n_devices).reshape(-1, 5)
ws = np.stack([w]*n_devices)

jax.pmap(convolve, in_axes=(None, 0))(x, ws)
```

Output: 

```
ShardedDeviceArray([[11., 20., 29.],
                    [11., 20., 29.],
                    [11., 20., 29.],
                    [11., 20., 29.],
                    [11., 20., 29.],
                    [11., 20., 29.],
                    [11., 20., 29.],
                    [11., 20., 29.]], dtype=float32)
```
